src/storage/checkpoint_manager.cpp

```diff
@@ -0,0 +1,77 @@
+#include "storage/checkpoint_manager.h"
+#include <vector>
+
+#define NUM_RESERVED_COLUMNS 1u
+
+namespace terrier::storage {
+void CheckpointManager::Checkpoint(const SqlTable &table, const BlockLayout &layout) {
+  std::vector<storage::col_id_t> all_col(layout.NumColumns() - NUM_RESERVED_COLUMNS);
+  // Add all of the column ids from the layout to the projection list
+  // 0 is version vector so we skip it
+  for (uint16_t col = NUM_RESERVED_COLUMNS; col < layout.NumColumns(); col++) {
+    all_col[col - NUM_RESERVED_COLUMNS] = storage::col_id_t(col);
+  }
+
+  // TODO(Mengyang): should calculate number of tuple to fit in buffer_size_
+  // now I only use a magic number here
+  uint32_t max_tuples = 10;
+
+  ProjectedColumnsInitializer column_initializer(layout, all_col, max_tuples);
```

Comment:
Instead of using the block layout that is passed in, you guys should use SqlTable::InitializerForProjectedColumns or SqlTable::InitializerForProjectedRow

Reply:
Thanks for the advice. We have changed it.